### PR TITLE
Ah gawb 1710: update permission checks in the ui

### PR DESCRIPTION
**Pull request by ahaessly**
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/page/workspace/summary/catalog/wizard.cljs [hunk 1]
```diff
@@ -171,15 +171,17 @@
                              :enumerate enumerate
                              :questions questions
                              :attributes working-attributes
-                             :editable? (and writer? curator?)
+                             :editable? (or writer? catandread?)
+                             :set-discoverable? (or can-share? catandread? owner?)
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/catalog/wizard.cljs]: This whole map is a prime candidate for my new `restructure` util.  `(utils/restructure foo bar baz)` gives `{:foo foo :bar bar :baz baz}`.

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 2]
```diff
@@ -90,8 +91,9 @@
                                                        :owner? owner?
                                                        :curator? curator?
                                                        :writer? writer?
+                                                       :catandread? catandread?
                                                        :request-refresh request-refresh}])}])
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs]: `restructure` here too

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 3]
```diff
@@ -230,7 +232,8 @@
                      :can-share? can-share?
                      :owner? owner?
                      :curator? curator?
-                     :writer? writer?}])
+                     :writer? writer?
+                     :catandread? catandread?}])
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs]: and here

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 4]
```diff
@@ -261,9 +264,10 @@
          (let [owner? (or (= "PROJECT_OWNER" (:accessLevel workspace)) (= "OWNER" (:accessLevel workspace)))
                writer? (or owner? (= "WRITER" (:accessLevel workspace)))
                can-share? (:canShare workspace)
+               catandread? (and (or writer? (reader? workspace)) (:catalog workspace))
                user-access-level (:accessLevel workspace)
                derived {:owner? owner? :writer? writer? :reader? (reader? (:workspace props))
-                        :can-share? can-share? :user-access-level user-access-level :request-refresh #(react/call :refresh this)}]
+                        :can-share? can-share? :catandread? catandread? :user-access-level user-access-level :request-refresh #(react/call :refresh this)}]
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs]: and yet again

File: src/cljs/main/broadfcui/page/workspace/summary/tab.cljs [hunk 5]
```diff
@@ -261,9 +264,10 @@
          (let [owner? (or (= "PROJECT_OWNER" (:accessLevel workspace)) (= "OWNER" (:accessLevel workspace)))
                writer? (or owner? (= "WRITER" (:accessLevel workspace)))
                can-share? (:canShare workspace)
+               catandread? (and (or writer? (reader? workspace)) (:catalog workspace))
```
Comment (MattPutnam) [inline src/cljs/main/broadfcui/page/workspace/summary/tab.cljs]: I'm not wild about this name, but I also can't think of anything better.

**Review by MattPutnam — APPROVED** [review]
👍, just some cleanup

**Review by davidangb — APPROVED** [review]
lgtm! did not test.

**Pull request merged**